plugins/config/src/configenv/configenv.go

```diff
@@ -0,0 +1,194 @@
+package configenv
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"archive/tar"
+
+	common "github.com/dokku/dokku/plugins/common"
+)
+
+//Env is a representation for global or app environment
+type Env struct {
+	name           string
+	env            map[string]string
+	filename       string
+	EscapeNewlines bool
+}
+
+func (e *Env) String() string {
+	return e.EnvfileString()
+}
+
+//EnvfileString returns the contents of this Env in ENVFILE format
+func (e *Env) EnvfileString() string {
```

Comment:
For env parsing, could we instead lean on something like [godotenv](https://github.com/joho/godotenv). Dotenv file parsing is hard, and now that we're at a higher-level language, might do some good to offload the munging to a proven library?